clippy_lints/src/matches/single_match.rs

```diff
@@ -249,7 +249,10 @@ impl<'a> PatState<'a> {
         let states = match self {
             Self::Wild => return None,
             Self::Other => {
-                *self = Self::StdEnum(cx.arena.alloc_from_iter((0..adt.variants().len()).map(|_| Self::Other)));
+                *self = Self::StdEnum(
+                    cx.arena
+                        .alloc_from_iter(std::iter::repeat_with(|| Self::Other).take(adt.variants().len())),
```

Comment:
Happy for me to implement (derive) clone for `PatState`?

Reply:
Ah, good point, I missed that it's not such a trivial change. I don't think it can implement Clone because it stores a mutable reference. Nevermind then.